### PR TITLE
UTCDate integration

### PR DESCRIPTION
**Pull request by kossnocorp**
- This integrates the `UTCDate` class (https://github.com/date-fns/utc) that replaces internal UTC functions making the build size smaller.
- It makes all functions that accept `Date` as an argument generic. This allows using any class that extends `Date`.
- It also updates TypeScript config and forces using `index` in imports

### CODE REVIEW

File: src/closestTo/index.ts [hunk 1]
```diff
@@ -23,9 +23,9 @@ import requiredArgs from '../_lib/requiredArgs/index'
  * ])
  * //=> Tue Jan 01 2030 00:00:00
  */
-export default function closestTo(
-  dirtyDateToCompare: Date | number,
-  dirtyDatesArray: Array<Date | number>
+export default function closestTo<DateType extends Date>(
+  dirtyDateToCompare: DateType | number,
+  dirtyDatesArray: Array<DateType | number>
 ): Date | undefined {
```
Comment (leshakoss) [inline src/closestTo/index.ts:29]: Missed the return type

File: src/max/index.ts [hunk 2]
```diff
@@ -23,7 +23,9 @@ import requiredArgs from '../_lib/requiredArgs/index'
  * ])
  * //=> Sun Jul 02 1995 00:00:00
  */
-export default function max(dirtyDatesArray: Array<Date | number>): Date {
+export default function max<DateType extends Date>(
+  dirtyDatesArray: Array<DateType | number>
+): DateType | Date {
```
Comment (leshakoss) [inline src/max/index.ts:28]: Can this be just `DateType`? Also in `min` and `clamp`

**Review by leshakoss — DISMISSED** [review]
What I found so far

File: src/max/index.ts (hunk 2 above)
Comment (kossnocorp) [reply src/max/index.ts:28]: Nope, if there's no elements, we'll return `new Date(NaN)`

**Comment by gkostov** [comment]
Hey guys, is this work going to be released soon? (hopefully as part of version 2)

As I understand it, it mostly improves the internal functionality of the library but luckily it happens to also open up the library for a solution that we're trying to come up with in relation to working with arbitrary timezones. So I'm looking for a way to get this down with an official release.

Cheers

**Pull request merged**